packages/web/hooks/use-all-pools-table.ts

```diff
@@ -0,0 +1,175 @@
+  const [page, setPage, minPage, numPages, allPoolsPages] = usePaginatedData(
+    sortedAllPoolsWithMetrics,
+    10
+  );
+  const tableCols = [
+    {
+      id: "pool.id",
+      display: "Pool ID/Tokens",
+      displayClassName: "!pl-[5.25rem]",
+      sort:
+        sortKeyPath === "pool.id"
+          ? {
+              currentDirection: sortDirection,
+              onClickHeader: toggleSortDirection,
+            }
+          : {
+              onClickHeader: () => {
+                setSortKeyPath("pool.id");
+                setSortDirection("ascending");
+              },
+            },
+      displayCell: PoolCompositionCell,
+    },
+    {
+      id: "liquidity",
+      display: "Liquidity",
+      infoTooltip: "This is liquidity",
+      sort:
+        sortKeyPath === "liquidity"
+          ? {
+              currentDirection: sortDirection,
+              onClickHeader: toggleSortDirection,
+            }
+          : {
+              onClickHeader: () => {
+                setSortKeyPath("liquidity");
+                setSortDirection("ascending");
+              },
+            },
+    },
+    {
+      id: "volume24h",
+      display: "Volume (24H)",
+      sort:
+        sortKeyPath === "volume24h"
+          ? {
+              currentDirection: sortDirection,
+              onClickHeader: toggleSortDirection,
+            }
+          : {
+              onClickHeader: () => {
+                setSortKeyPath("volume24h");
+                setSortDirection("ascending");
+              },
+            },
+
+      displayCell: MetricLoaderCell,
+    },
+    {
+      id: "fees7d",
+      display: "Fees (7D)",
+      sort:
+        sortKeyPath === "fees7d"
+          ? {
+              currentDirection: sortDirection,
+              onClickHeader: toggleSortDirection,
+            }
+          : {
+              onClickHeader: () => {
+                setSortKeyPath("fees7d");
+                setSortDirection("ascending");
+              },
+            },
+      displayCell: MetricLoaderCell,
+    },
+    {
+      id: isIncentivizedPools ? "apr" : "myLiquidity",
+      display: isIncentivizedPools ? "APR" : "My Liquidity",
+      sort:
+        sortKeyPath === (isIncentivizedPools ? "apr" : "myLiquidity")
+          ? {
+              currentDirection: sortDirection,
+              onClickHeader: toggleSortDirection,
+            }
+          : {
+              onClickHeader: () => {
+                setSortKeyPath(isIncentivizedPools ? "apr" : "myLiquidity");
+                setSortDirection("ascending");
+              },
+            },
+      displayCell: isIncentivizedPools ? MetricLoaderCell : undefined,
+    },
+  ];
+  useEffect(() => {
+    setSortKeyPath("liquidity");
+    setSortDirection("descending");
+  }, []);
+  const baseRow: RowDef = {
+    makeHoverClass: () => "text-secondary-200",
+  };
+
+  const tableRows: RowDef[] = pools.map(() => ({
+    ...baseRow,
+    onClick: (i) => console.log(i),
```

Comment:
How about change it as a link using `next/link`, not `onClick`? It would be better for SEO and when user want to open pool detail page with a new tab.

Reply:
Ah yeah. Good idea, hadn't occurred to me.